e33lin/grocery-store
# Viewing a Recommendation

## Summary
- stores recommendations into the Recommendations table
- populates info for /recommendations/:id
- checks the user session so that when the same user goes back to /recommendations, we don't continue to add the same recs into the Recommendations table

## Testing
BEFORE TESTING: Please change the`session[:current_user_id] = 1234567` in the recommendations controller to some random set of numbers (the point of this is to just distinguish your own "recommendations" from those that already exist in the db by setting your own "session id")

1. go to /recommendations on the app. you should see successful insertions into the database in your terminal. please note that the list_id should match whatever session id you just set:
![Screen Shot 2023-01-29 at 5 28 04 PM](https://user-images.githubusercontent.com/65259395/215359202-9799e403-9edb-41ab-9fe6-6967e2ccfc56.png)
2. click on any of the recommendations on the page. it should take you to /recommendations/:id (ex., "store1" link takes you to /recommendations/1). it should also populate the correct information based on the fake data (see below).
![Screen Shot 2023-01-29 at 5 30 58 PM](https://user-images.githubusercontent.com/65259395/215359329-97c3fddb-b95e-4f2f-8e2d-12f1d6850875.png)
3. return back to the list of recommendations. please check that you aren't inserting more recommendations into the db, but just querying on the ones you inserted in step 1:
![Screen Shot 2023-01-29 at 5 33 37 PM](https://user-images.githubusercontent.com/65259395/215359444-3ded222b-1ab7-4ed9-9117-c7eba5bc47b7.png)

## Notes
The data here is not live!! its just a json object with faux information. you can see this object in the `print_dict.py` file:
![Screen Shot 2023-01-29 at 5 36 18 PM](https://user-images.githubusercontent.com/65259395/215359673-a1bc6f17-1e3b-4d69-abe5-928978e5a542.png)
i was just using the example stuff in the tech doc (thank you @caitlan-krasinski ) just to get this functionality to work for now, but i will update the actual parsing and stuff to match the real json object structure in another PR. 

im thinking for sessions, we can just set a random generated number when the user reaches /home, and purge it (along with their list and their recommendations) when they close browser. when generating the random number, we can just use a really big range (0-100000), and double check if that session is already being used by querying for it like `List.find_by(list_id: session_id)`. if it turns up empty, great! if not, then generate a new random number. what do you all think?
